main/resources/teamcity/utils.kt

```diff
@@ -32,6 +32,22 @@ fun Platform.buildTypeId(): String = buildTypeName().substringBefore(" ")
 fun Platform.teamcityAgentName(): String = buildTypeName()
 
 
+const val BUILD_CONFIGURE_VERSION_ID = "Build_Version"
+const val BUILD_ALL_ID = "Build_All"
+const val DEPLOY_CONFIGURE_VERSION_ID = "Deploy_Configure"
+const val DEPLOY_PUBLISH_ID = "Deploy_Publish"
+
+private fun existingBuildId(suffix: String): String = "RootProjectId_$suffix"
+private fun Project.existingBuildWithId(id: String): BuildType = buildTypes.single { it.id.toString() == existingBuildId(id) }
+
+fun Project.existingBuildVersion(): BuildType = existingBuildWithId(BUILD_CONFIGURE_VERSION_ID)
```

Comment:
The `existing` prefix sounds quite generic.

Perhaps we can get a DSL like that:
- `Project.knownBuilds` property that returns `KnownBuilds` type that is a wrapper around `Project`
- then `KnownBuilds` defines `BuildType` properties: `buildVersion`, `buildAll`, `buildOn(platform)` etc.

The usages in `additionalConfiguration` would look like:
`knownBuilds.buildVersion` or `knownBuilds.buildAll`

Reply:
done